### PR TITLE
Make code Python3.6 compatible and add verify data method

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: resolwe/storage/models.py [hunk 1]
```diff
@@ -243,6 +246,44 @@ def transfer_data(self, destination: "StorageLocation"):
...
+                        )
+                    )
+                    continue
+
+                # Hashes differ: log error and continue to check other hashes.
+                if referenced_path != hash_value:
+                    return_value = False
+                    logger.error(
+                        "ReferencedPath with id {} has wrong {} hash: {} instead of {}".format(
+                            referenced_path.id,
```
Comment (dblenkus) [inline resolwe/storage/models.py:278]: The string has 4 placeholders and 5 values to insert.

File: resolwe/storage/models.py [hunk 2]
```diff
@@ -243,6 +246,44 @@ def transfer_data(self, destination: "StorageLocation"):
...
+        and cause network traffic, use with care.
+
+        :returns: True if data is OK, False otherwise. In case of failure
+        additional information about error is logged.
+        """
+        hash_types = ["md5", "crc32c", "awss3etag"]
+        return_value = True
+        for referenced_path in self.files:
+            url = self.get_path(filename=referenced_path.path)
+            hashes = dict(zip(hash_types, self.connector.get_hashes(url, hash_types)))
```
Comment (dblenkus) [inline resolwe/storage/models.py:262]: Is there any use-case for `get_hashes` where you don't need such dict? If not, the method can return it.

File: resolwe/storage/models.py (hunk 1 above)
Comment (gregorjerse) [reply resolwe/storage/models.py:288]: Fixed.

File: resolwe/storage/models.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/storage/models.py]: Not with new `transfer_objects` method, except one migration. I have changed method signature and the corresponding code.

File: resolwe/storage/connectors/baseconnector.py [hunk 3]
```diff
@@ -135,6 +127,15 @@ def after_push(self, objects: List[dict], url: str) -> Optional[List[dict]]:
         """
         return objects
 
+    @validate_url
+    def check_url(self, url: Union[str, PathLike]) -> bool:
```
Comment (dblenkus) [inline resolwe/storage/connectors/baseconnector.py:131]: Squash this commit with the previous one.

**Pull request merged**